tests/loan_protocol.rs

```diff
@@ -472,22 +380,38 @@ async fn lend_and_dynamic_liquidate() {
 
     let borrower = borrower.interpret(SECP256K1, loan_response).unwrap();
     let loan_transaction = borrower
-        .sign(|transaction| async move { Ok(borrower_wallet.sign_all_inputs(transaction)) })
+        .sign({
+            let wallet = wallet.clone();
+            |tx| async move {
+                let wallet = wallet.lock().unwrap();
+                Ok(wallet.sign_inputs(tx))
+            }
+        })
         .await
         .unwrap();
 
     let loan_transaction = lender
         .finalise_loan(loan_transaction, {
-            let client = client.clone();
-            |transaction| async move { client.sign_raw_transaction(&transaction).await }
+            let wallet = wallet.clone();
+            |tx| async move {
+                let wallet = wallet.lock().unwrap();
+                Ok(wallet.sign_inputs(tx))
+            }
         })
         .await
         .unwrap();
 
-    client
-        .send_raw_transaction(&loan_transaction)
-        .await
-        .unwrap();
+    let utxos = {
+        let wallet = wallet.lock().unwrap();
+        wallet
+            .utxos
+            .iter()
+            .map(|(input, _, _)| input.clone())
+            .collect::<Vec<_>>()
+    };
+
+    verify_wallet_inputs_spend(&loan_transaction, &utxos)
```

Comment:
If you intend to always pass the wallet's UTXOs in, you could make this a function on `Wallet`.